### PR TITLE
Create verticalLine.lua

### PR DESCRIPTION
**Pull request by ascpial**
Just a question about copyright, I obviously accept that my contribution is licensed with the CC0 and accept that my contribution is given to the public domain, but do I need to add my name in the license header in the `utils.lua` file?

### CODE REVIEW

**Comment by hooded-person** [comment]
nice

File: verticalLine.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+function PrimeUI.verticalLine(win, x, y, height, side, fgColor, bgColor)
+    expect(1, win, "table")
+    expect(2, x, "number")
+    expect(3, y, "number")
+    expect(4, height, "number")
+    side = expect(5, side, "string", "nil") or "left"
+    fgColor = expect(6, fgColor, "number", "nil") or colors.white
+    bgColor = expect(7, bgColor, "number", "nil") or colors.black
+    -- Use drawing characters to draw a thin line.
+    win.setTextColor(side=="left" and fgColor or bgColor)
```
Comment (MCJack123) [inline verticalLine.lua]: Please adjust the spacing and indentation to match the style of the rest of the code.

File: verticalLine.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    win.setBackgroundColor(side=="left" and bgColor or fgColor)
+    for j=1, height do
+      win.setCursorPos(x, y+j-1)
+      win.write(("\x95"))
```
Comment (MCJack123) [inline verticalLine.lua]: There's an extra set of parentheses that should be removed.

File: verticalLine.lua [hunk 3]
```diff
@@ -0,0 +1,28 @@
+local PrimeUI = require "util" -- DO NOT COPY THIS LINE
+local expect = require "cc.expect".expect -- DO NOT COPY THIS LINE
+-- Start copying below this line. --
+
+--- Draws a vertical line at a position with the specified height.
+---@param win window The window to draw on
+---@param x number The X position of the left side of the line
+---@param y number The Y position of the line
+---@param height number The width/length of the line
+---@param side "right"|"left"|nil THe side of the line, either right or left (defaults to left)
```
Comment (MCJack123) [inline verticalLine.lua]: Could you make this a boolean instead? There's only two possibilities, so using a string is a bit overcomplicated.

File: verticalLine.lua [hunk 4]
```diff
@@ -0,0 +1,28 @@
+local PrimeUI = require "util" -- DO NOT COPY THIS LINE
+local expect = require "cc.expect".expect -- DO NOT COPY THIS LINE
+-- Start copying below this line. --
+
+--- Draws a vertical line at a position with the specified height.
+---@param win window The window to draw on
+---@param x number The X position of the left side of the line
```
Comment (MCJack123) [inline verticalLine.lua]:
```suggestion
---@param x number The X position of the line
```

File: verticalLine.lua [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+---@param y number The Y position of the line
```
Comment (MCJack123) [inline verticalLine.lua]:
```suggestion
---@param y number The Y position of the top of the line
```

File: verticalLine.lua [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+---@param height number The width/length of the line
```
Comment (MCJack123) [inline verticalLine.lua]:
```suggestion
---@param height number The height of the line
```

**Review by MCJack123 — CHANGES_REQUESTED** [review]
Looks alright, just needs a few style fixes before merging.

**Comment by hooded-person** [comment]
Maybe also updated the singular releases file, it is currently already missing the monitor support and the whole clickArea item

**Review by MCJack123 — APPROVED** [review]
Made a few minor changes and a syntax fix, now it LGTM :+1:

**Pull request merged**